### PR TITLE
Add 'after'Action methods, to allow addOrderMenus to be run

### PR DESCRIPTION
**Pull request by ryanmitchell**
How is this approach, I had to add new methods to RestController to allow it.

See also:
https://github.com/tastyigniter/TastyIgniter/pull/410
https://github.com/tastyigniter/ti-ext-cart/pull/10

### CODE REVIEW

File: actions/RestController.php [hunk 1]
```diff
@@ -205,6 +215,56 @@ public function restFindModelObject($recordId)
...
         return $result;
     }
+    
+    /**
+     * Run logic after index operation
+     * by overriding it in the controller.
+     * @param \Model $model
+     * @return \Model
+     */
+    public function afterIndex($model)
```
Comment (sampoyigi) [inline actions/RestController.php:225]: Prefix these new methods with `rest` for context.

File: apiresources/Orders.php [hunk 2]
```diff
@@ -40,4 +40,25 @@ public function store()
 
         $this->asExtension('RestController')->store();
     }
+    
+    public function afterStore($model)
+    {
+        if (($menu_items = Request::input('menu_items', []) && count($menu_items))
```
Comment (sampoyigi) [inline apiresources/Orders.php:46]: camelCase variable names here

Comment (sampoyigi) [reply apiresources/Orders.php:46]: and it needs to be wrapped properly as not to get unexpected results from the if statement

File: apiresources/Orders.php [hunk 3]
```diff
@@ -40,4 +40,25 @@ public function store()
...
+        if (($token = $this->getToken()) && $token->isForCustomer())
+            Request::merge(['customer_id' => $token->tokenable_id]);
+
+        $this->asExtension('RestController')->update();
+        
+    }
+    
+    public function afterUpdate($model)
+    {
+        if (($menu_items = Request::input('menu_items', []) && count($menu_items))
```
Comment (sampoyigi) [inline apiresources/Orders.php:61]: same with this if statement

**Comment by ryanmitchell** [comment]
All updated

**Comment by sampoyigi** [comment]
Made some changes to the lifecycle methods to use the same approach used in the FormController, let me know what you think before I merge

**Comment by ryanmitchell** [comment]
Yep looks good and makes sense

**Pull request merged**